matching/views.py

```diff
@@ -35,15 +39,20 @@ def show_potential_matches(request, utrip_id):
         start_trip__lte=current_usertrip.end_trip,
         end_trip__gte=current_usertrip.start_trip,
         travel_type=current_usertrip.travel_type,
-        trip=current_usertrip.trip,
+        trip=current_usertrip.trip,  # destination matching
         user__is_active=True,
         user__userprofile__dob__lte=date.today()
         - timedelta(days=365.25 * current_user.userprofile.age_lower),
         user__userprofile__dob__gt=date.today()
         - timedelta(days=365.25 * current_user.userprofile.age_upper + 1),
     )
 
-    try:
+    # language filter that filters users with at least one common language
+    # apply this filter only if the current user has any language preferences
+    if len(current_user.userprofile.languages) != 0:
```

Comment:
when we create user profile now, it doesn't let you go forward without picking a language - does this still apply in that case